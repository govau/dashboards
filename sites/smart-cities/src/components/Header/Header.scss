@import '~style/main';

.header {
  background: $color-primary--600;
  color: $color-white;
}

// this is the minor header on desktop size
// and the only header on mobile/tablet
.govContainer {
  display: inline-flex;
  position: relative;
  z-index: $z-index-header;
  height: $main-header-height-mobile;
  flex-flow: column wrap;
  align-items: flex-start;
  align-content: flex-start;
  justify-content: center;

  @include for-desktop-up {
    display: flex;
    padding-left: 30px;
    height: $gov-header-height;
    align-items: center;
    flex-flow: row nowrap;
    justify-content: flex-start;
    background: $color-primary--500;

    > * {
      margin-right: 8px;
    }
  }
}

.compactTitle,
.compactSubtitle {
  @include for-desktop-up {
    display: none;
  }
}

.govAu,
.govWebsite {
  display: none;

  @include for-desktop-up {
    display: inline;
  }
}

.govAu {
  font-size: 14px;
}

.govWebsite {
  font-size: 12px;
}

.mainContainer {
  position: absolute;
  right: 4px;
  top: 12px;

  @include for-desktop-up {
    position: relative;
    z-index: $z-index-header;
    display: flex;
    width: 100%;
    height: $main-header-height-desktop;
    justify-content: space-between;
  }
}

.titleContainer,
.orgContainer {
  display: inline-flex;
  flex-direction: column;

  @include for-desktop-up {
    padding: 0 32px;
  }
}

.orgContainer {
  align-items: flex-end;
}

.titleContainer,
.orgLink {
  display: none;

  @include for-desktop-up {
    display: block;
  }
}

.star {
  display: none;

  @include for-desktop-up {
    display: block;
    height: 18px;
    width: 18px;
    flex: 0 1 auto;
  }
}

.hamburger {
  margin-right: 8px;
  margin-left: 8px;

  @include for-desktop-up {
    display: none;
  }
}

.projectStatus {
  margin-top: 12px;
  background: $color-white;
  align-self: flex-start;
  color: $color-primary--600;
  padding: 0 10px;
  font-size: 11px;
  font-weight: 500;
}

.titleLinkWrapper {
  text-decoration: none;
}

.title,
.subtitle {
  font-size: 36px;
  font-weight: 300;
  line-height: 38px;
}

.title {
  margin-top: 6px;
}

.compactTitle {
  font-size: 14px;
  line-height: 14px;
  flex: 0 0 16px;
}

.compactSubtitle {
  font-size: 14px;
  line-height: 14px;
  font-weight: 300;
  flex: 0 0 14px;
}

.orgLink {
  margin-top: 14px;
  margin-bottom: 16px;
  line-height: 1;
}

.orgLogo {
  height: 45px;
  width: 73px;
}

.contactLink {
  border-bottom: 1px solid $color-highlight;
  font-size: 16px;
  font-weight: 300;
  text-decoration: none;
}
