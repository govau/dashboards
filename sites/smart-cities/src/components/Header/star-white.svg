<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg width="22px" height="22px" viewBox="0 0 22 22" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 46.2 (44496) - http://www.bohemiancoding.com/sketch -->
    <title>star</title>
    <desc>Created with Sketch.</desc>
    <defs></defs>
    <g id="Symbols" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="hamburger-and-brand" transform="translate(-44.000000, 0.000000)" fill="#FFFFFF">
            <g id="Group-2" transform="translate(44.000000, 0.000000)">
                <g id="Group-3">
                    <polygon id="star" points="10.999599 0 8.55257018 6.35863332 2.17794386 4.35676524 5.5 10.2846948 0 14.147466 6.58877142 15.1810726 6.10433832 22 10.999599 17.3600717 15.8952607 22 15.4116296 15.1810726 22 14.147466 16.500401 10.2846948 19.8220561 4.35676524 13.4478308 6.35863332"></polygon>
                </g>
            </g>
        </g>
    </g>
</svg>
