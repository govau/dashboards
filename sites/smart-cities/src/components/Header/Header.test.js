import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

jest.mock('../CategoryNavContainer/CategoryNavContainer', () => 'CategoryNavContainer');
jest.mock('../CityHeader/CityHeader', () => 'CityHeader');
jest.mock('../HamburgerButton/HamburgerButton', () => 'HamburgerButton');
jest.mock('../Pill/Pill', () => 'Pill');

const openNavMock = jest.fn();

const defaultProps = {
  cityId: 'perth',
  categoryId: 'jobs',
  openNav: openNavMock,
};

it('should match Snapshot', () => {
  const component = shallow(
    <Header {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should open the nav when the hamburger is clicked', () => {
  const component = shallow(
    <Header {...defaultProps} />
  );

  component.find('HamburgerButton').simulate('click');

  expect(openNavMock).toHaveBeenCalledTimes(1);
});
