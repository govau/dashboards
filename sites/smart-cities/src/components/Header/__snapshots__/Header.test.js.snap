// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<header className=\\"header\\">
  <div className=\\"govContainer\\">
    <HamburgerButton className=\\"hamburger\\" onClick={[Function]} label=\\"open navigation menu\\" />
    <img className=\\"star\\" src=\\"star-white.svg\\" alt=\\"star icon\\" />
    <span className=\\"govAu\\">
      GOV.AU
    </span>
    <span className=\\"govWebsite\\">
      Official Australian Government Website
    </span>
    <div className=\\"compactTitle\\">
      National Cities Performance
    </div>
    <div className=\\"compactSubtitle\\">
      Framework Dashboard
    </div>
  </div>
  <div className=\\"mainContainer\\">
    <div className=\\"titleContainer\\">
      <Pill height={16} className=\\"projectStatus\\">
        ALPHA
      </Pill>
      <NavLink className=\\"titleLinkWrapper\\" to=\\"/all-cities/overview\\" activeClassName=\\"active\\" ariaCurrent=\\"true\\">
        <h1 className=\\"title\\">
          National Cities Performance
        </h1>
      </NavLink>
      <div className=\\"subtitle\\">
        Framework Dashboard
      </div>
    </div>
    <div className=\\"orgContainer\\">
      <a className=\\"orgLink\\" href=\\"https://dta.gov.au/\\">
        <img className=\\"orgLogo\\" src=\\"dta-logo.svg\\" alt=\\"DTA logo\\" />
      </a>
      <FeedbackButton />
    </div>
  </div>
  <CategoryNavContainer categoryId=\\"jobs\\" cityId=\\"perth\\" cityName={[undefined]} />
</header>"
`;
